File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.js
var evalRPN = function (tokens) {
  const operators = {
    "+": (a, b) => a + b,
    "-": (a, b) => a - b,
    "*": (a, b) => a * b,
    "/": (a, b) => Math.trunc(a / b),  // truncate toward zero
  };

  const stack = [];

  for (let token of tokens) {
    if (!(token in operators)) {
      stack.push(Number(token)); // ✅ 轉成數字再 push
    } else {
      const b = stack.pop();
      const a = stack.pop();
      const res = operators[token](a, b);
      stack.push(res);
    }
  }

  return stack.pop();
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in tokens:
            if i == "+":
                stack.append(int(stack.pop()) + int(stack.pop()))
            elif i == "-":
                right,left = int(stack.pop()), int(stack.pop())
                stack.append(left - right)
            elif i == "*":
                stack.append(int(stack.pop()) * int(stack.pop()))
            elif i == "/":
                right,left = int(stack.pop()),int(stack.pop())
                stack.append(int(left / right))
            else:
                stack.append(i)
        return int(stack[0])